src/main/java/org/apache/maven/plugins/gpg/GpgVersion.java

```diff
@@ -32,54 +32,46 @@
  */
 public class GpgVersion implements Comparable<GpgVersion>
 {
-    private final String rawVersion;
 
-    private GpgVersion( String rawVersion )
-    {
-        this.rawVersion = rawVersion;
-    }
+    private static final Pattern VERSION_PATTERN = Pattern.compile( "(\\d+\\.)+(\\d+)" );
 
-    public static GpgVersion parse( String rawVersion )
+    private final int[] versionSegments;
+
+    private GpgVersion( int... versionSegments )
     {
-        return new GpgVersion( rawVersion );
+        this.versionSegments = versionSegments;
     }
 
-    @Override
-    public int compareTo( GpgVersion other )
+    public static GpgVersion parse( String rawVersion )
```

Comment:
Where does this version string come from? That is, what is it the version of? Is it ever anything that does not match this pattern? E.g. 3.45.2-beta?

Reply:
As far as I can see the original author of this class made the assumption that it will always be a sequence of numbers delimited by dots (although the original author wrote in the class-level comment, that this is a SemVer parser).
The raw version string is read from the output of `gpg --version` apparently.

Unfortunately I am in no position to evaluate whether GPG will continue using this versioning scheme, but it seems they have consistently used it up until now: https://files.gpg4win.org/
In my opinion we can make the assumption, that GPG will continue using a three segment wide versioning scheme just as the original author implemented it.